src/main/java/com/google/sps/servlets/EventServlet.java

```diff
@@ -65,7 +76,87 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        // ToDo: Mingyi implment this method.
-        response.getWriter().println("You've called the post request");
+        String jsonBody = new BufferedReader(new InputStreamReader(request.getInputStream())).lines().collect(
+            Collectors.joining("\n"));
+
+        // Error handling: return error if jsonBody is empty
+        if (jsonBody == null || jsonBody.trim().length() == 0) {
+            System.out.println("the String fails to be parsed");
+            throw new NullPointerException();
+        }
+
+        // Parse the jsonBody String to a Gson object
+        Gson gson = new Gson();
+        Type type = new TypeToken<Map<String, String>>(){}.getType();
+        Map<String,String> eventJsonAsMap = gson.fromJson(jsonBody, type);
+
+        // Check if all the fields are present
+        if (!checkAllEventFieldsPresent(eventJsonAsMap)){
+            // throw error
+            System.out.println("Not all the fields in the map are present");
+            throw new NullPointerException();
+        }
+
+        // Get the value entered in the form  
+        Double event_Lat_D = Double.parseDouble(eventJsonAsMap.get("lat"));
+        Double event_Lng_D = Double.parseDouble(eventJsonAsMap.get("lng"));
+        String event_name = eventJsonAsMap.get("name");
```

Comment:
nit: I recommend using Jsoup to process the events properties that are stored as strings (to ensure that they are safe to display in HTML). You can see an example here: https://github.com/katherine-yu/software-product-sprint/blob/main/portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java.